### ISSUE TITLE
Functions taking in subset TypedDicts

### ISSUE THREAD

**Issue by theahura** [google/pytype#1254]
I have two TypedDicts. They share the same fields, except the second one has a few extra fields. I want to create a function that works on the former, that I can call with the latter.

Is there a way to make something like the following work?

```python
import typing

class Parent(typing.TypedDict):
  a: typing.Text

class Child(typing.TypedDict):
  a: typing.Text
  b: typing.Text

def baz(foo: Parent) -> int:
  return 0

x: Child = {'a': 'hello', 'b': 'world'}
baz(x)
```

This currently fails with the following:
```
Function baz was called with the wrong arguments [wrong-arg-types]                
         Expected: (foo: Parent)                                                                                                                           
  Actually passed: (foo: Child)         
                                                                            
  TypedDict extra keys: b
```

Having this functionality would be very useful for creating polymorphic abstract functions that can plug into a bunch of different data structures

**Comment by rchen152** [comment]
@martindemello What do you think? Does it seem reasonable to allow this? The type consistency rules in https://peps.python.org/pep-0589/#type-consistency are sort of confusing, but I *think* Child here should be considered consistent with Parent based on this section:

> a TypedDict type A is consistent with TypedDict B if A is structurally compatible with B. This is true if and only if both of these conditions are satisfied:
> * For each key in B, A has the corresponding key and the corresponding value type in A is consistent with the value type in B. For each key in B, the value type in B is also consistent with the corresponding value type in A.
> * For each required key in B, the corresponding key is required in A. For each non-required key in B, the corresponding key is not required in A.

**Comment by martindemello** [comment]
i agree, it does seem like a superset typed dict should be considered a subclass of the parent dict.

**Comment by martindemello** [comment]
typing-sig thread on the issue: https://mail.python.org/archives/list/typing-sig@python.org/thread/66RITIHDQHVTUMJHH2ORSNWZ6DOPM367/
